thinc/layers/with_flatten_v2.py

```diff
@@ -0,0 +1,67 @@
+from typing import Tuple, Callable, Sequence, Any, cast, TypeVar, Optional, List
+
+from ..model import Model
+from ..config import registry
+
+
+InItemT = TypeVar("InItemT")
+OutItemT = TypeVar("OutItemT")
+ItemT = TypeVar("ItemT")
+
+NestedT = List[List[ItemT]]
+FlatT = List[ItemT]
+
+
+@registry.layers("with_flatten.v2")
+def with_flatten_v2(
+    layer: Model[FlatT[InItemT], FlatT[OutItemT]]
+) -> Model[NestedT[InItemT], NestedT[OutItemT]]:
+    return Model(f"with_flatten({layer.name})", forward, layers=[layer], init=init)
+
+
+def forward(
+    model: Model[NestedT[InItemT], NestedT[OutItemT]],
+    Xnest: NestedT[InItemT],
+    is_train: bool,
+) -> Tuple[NestedT[OutItemT], Callable]:
+    layer: Model[FlatT[InItemT], FlatT[OutItemT]] = model.layers[0]
+    Xflat = _flatten(Xnest)
+    Yflat, backprop_layer = layer(Xflat, is_train)
+    # Get the split points. We want n-1 splits for n items.
+    lens = [len(x) for x in Xnest]
```

Comment:
Could the `_flatten` return the `lens` to not do the for loop over `Xnest` twice?

Reply:
Done!